feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -136,6 +158,77 @@ class PreviewViewModel @AssistedInject constructor(
         }
     }
 
+    /**
+     * Returns the difference between two [CameraAppSettings] as a queue of
+     * [Pair]<[SettingNames], [Any]>.
+     */
+    private fun getSettingsDiff(
+        oldCameraAppSettings: CameraAppSettings,
+        newCameraAppSettings: CameraAppSettings
+    ): ArrayDeque<Pair<KProperty<Any?>, Any?>> {
+        val queue = ArrayDeque<Pair<KProperty<Any?>, Any?>>()
+        CameraAppSettings::class.memberProperties.forEach { property ->
+            if (property.get(oldCameraAppSettings) != property.get(newCameraAppSettings)) {
+                queue.add(Pair(property, property.get(newCameraAppSettings)))
+            }
+        }
+        return queue
+    }
+
+    /**
+     * Iterates through a queue of [Pair]<[KProperty], [Any]> and attempt to apply them to
+     * [CameraUseCase].
+     */
+    private suspend fun applySettingsDiff(
+        newSettingsDiff: ArrayDeque<Pair<KProperty<Any?>, Any?>>
```

Comment:
If changing this argument to a `Map`, just change the name of `pair` below to `entry` and change `pair.first` -> `entry.key` and `pair.second` -> `entry.value`.